### DISCUSSION TITLE
Cityscape powered by analytics

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#1140] [Ideas]
https://github.com/lowlighter/metrics/pull/1139 and https://github.com/honzaap/GitHubCity got me thinking there is some value to this skyline idea, beyond just the pretties.  That project poses the question:
> What would Your city look like?

Provocative.  I wonder if that project really intends to answer the question, because personalising each persons city would depend on the type of analytics being done by this project.

Anyway, the ideas that immediately came to mind:
- Different building materials/shapes/colors for different programming languages used during the day.
- Branding on top of the buildings, either the repo or the language logo, if there are large enough contribs in a single day so warrant the entire tower bring branded. Blinking neon lights! Without them, a city is having a power grid failure!
- It seems zooming already allows a quite clear picture of each level of each building, so branding could be put on each level of each building, if the data justifies it
- Merge towers together into larger buildings if the majority of multiple days contribs were for the same project (I see GitHubCity is already merging cells to form yellow low-rise buildings, and I cant imagine what data the algo is using to make that decision)
- Add overhead walkways between nearby tall but disjoint towers that were predominately for the same project.
- Incoming and outgoing container ships showing type/volume/etc of incoming and outgoing contributions of various types (PRs, reviews, issues, discussions, etc).
- When adjoining days with zero contribs are surrounded by towers, I think those 0 contrib days should be merged together to create larger park-land sections, as it typically means the user went on holidays. This merging into park-land wont make sense for the part of the year which is still in the future. Perhaps some building equipment could be put on the future few weeks, to indicate those weeks are "under construction", and then park-land for future weeks ahead of that. (idea also proposed at https://github.com/honzaap/GithubCity/issues/18)

How to marry together these two projects is an interesting problem.  Ping @honzaap in case you are interesting in this.

**Comment by lowlighter** [comment]
While these are good ideas and would love to see something that creative and artistic, I feel like it's too much work to put on a few people 😅 

Most of the work would probably be on [GitHub City](https://github.com/honzaap/GithubCity) project anyway (3D models, animations, merging algo, etc.) as fetching data from API is easier I think. Maybe if GitHub City gains more visibility and traction it could happen though

Also metrics currently doesn't have an "output schema" which makes it harder to interface it with other applications 😕
